src/app/helpers/functools.js

```diff
@@ -416,6 +417,20 @@ const memoizeWith = (keyFn, fn) => {
   };
 };
 
+const memoizeOne = fn => {
+  let key;
+  let cache;
+  return (...args) => {
+    if (key === args[0]) {
+      return cache;
+    } else {
+      key = args[0];
+      cache = fn(...args);
+      return cache;
+    }
+  };
+};
+
```

Comment:
Das ist im Prinzip eine naive Neuimplementierung von `React.useMemo`.

```
   const cached1 = memoizeOne((a1, ..., aN) => calcValue...);
   // entspricht
   const cached2 = useMemo((a1, ..., aN) => calcValue..., [a1])
```

Natürlich kann man den Hook nur in einer Funktionskomponente nutzen...